application/src/main/java/org/togetherjava/tjbot/commands/tags/TagSystem.java

```diff
@@ -2,23 +2,38 @@
 
 import net.dv8tion.jda.api.entities.Emoji;
 import net.dv8tion.jda.api.interactions.callbacks.IReplyCallback;
+import net.dv8tion.jda.api.interactions.commands.OptionMapping;
+import net.dv8tion.jda.api.interactions.components.ActionRow;
 import net.dv8tion.jda.api.interactions.components.buttons.Button;
 import net.dv8tion.jda.api.interactions.components.buttons.ButtonStyle;
+import org.apache.commons.collections4.ListUtils;
 import org.togetherjava.tjbot.commands.utils.StringDistances;
 import org.togetherjava.tjbot.db.Database;
 import org.togetherjava.tjbot.db.generated.tables.Tags;
 import org.togetherjava.tjbot.db.generated.tables.records.TagsRecord;
 
+import javax.annotation.Nullable;
 import java.awt.Color;
-import java.util.Optional;
-import java.util.Set;
+import java.util.*;
+import java.util.function.Function;
 import java.util.stream.Collectors;
 
 /**
  * The core of the tag system. Provides methods to read and create tags, directly tied to the
  * underlying database.
  */
 public final class TagSystem {
+    /**
+     * The amount of candidates that are suggested as buttons in the message that tells the user
+     * that the tag they tried accessing doesn't exist
+     */
+    private static final int CANDIDATE_SUGGESTIONS_PER_USAGE = 5;
```

Comment:
I dont like the name of this. maybe `TAG_SUGGESTIONS_AMOUNT`

Reply:
done